File: 206-reverse-linked-list/206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseList(self, head):
        # in-place solution 
        if head == None or head.next == None:
            return head
        
        current = head
        previous_node = None
        next_node = head
        
        while current != None:
            next_node = current.next
            current.next = previous_node
            previous_node = current
            current = next_node
            
        return previous_node

File: 206-reverse-linked-list/206-reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    if (head == null || head.next == null){
        return head
    }
 
    let current = head
    let previous_node = null
    let next_node = head
        
    while (current != null){
        next_node = current.next
        current.next = previous_node
        previous_node = current
        current = next_node
    }
            
    return previous_node
};